framework.go

```diff
@@ -0,0 +1,275 @@
+package clustertest
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/giantswarm/clustertest/pkg/application"
+	"github.com/giantswarm/clustertest/pkg/client"
+	"github.com/giantswarm/clustertest/pkg/organization"
+	"github.com/giantswarm/clustertest/pkg/wait"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	applicationv1alpha1 "github.com/giantswarm/apiextensions-application/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+	cr "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	KubeconfigEnvVar = "E2E_KUBECONFIG"
+)
+
+// Framework is the overall framework for testing of clusters
+type Framework struct {
+	DisableLogging bool
+
+	mcKubeconfigPath string
+	mcClient         *client.Client
+	wcClients        map[string]*client.Client
+}
+
+// New initializes a new Framework instance using the kubeconfig found in the env var `E2E_KUBECONFIG`
+func New() (*Framework, error) {
+	mcKubeconfig, ok := os.LookupEnv(KubeconfigEnvVar)
+	if !ok {
+		return nil, fmt.Errorf("no %s set", KubeconfigEnvVar)
+	}
+
+	framework := NewWithKubeconfig(mcKubeconfig)
+
+	// for _, envVar := range os.Environ() {
+	// 	if strings.HasPrefix(envVar, "E2E_WC_") {
+	// 		// TODO: Initialize workload cluster
+	// 	}
+	// }
+
+	return framework, nil
+}
+
+// NewWithKubeconfig generates a new framework initialised with the Management Cluster provided as a Kubeconfig
+func NewWithKubeconfig(kubeconfigPath string) *Framework {
+	mcClient, err := client.New(kubeconfigPath)
+	if err != nil {
+		panic(err)
```

Comment:
Nitpick: I would rather return the error I think. The output would be nicer (and you'd immediately know what you did wrong) if you check the error wherever this is called. You can also call `ginkgo.Fail("message")` here but I suppose you don't want to couple this package to ginkgo.

Reply:
Yeah, makes sense. I'll update.

Reply:
Done!